sys/net/network_layer/fib/fib.c

```diff
@@ -609,6 +691,712 @@ int fib_get_num_used_entries(void)
+    vtimer_now(&now);
+
+    mutex_lock(&mtx_access);
+    if (fib_sr == NULL || addr == NULL || sr_path_entry == NULL) {
+        mutex_unlock(&mtx_access);
+        return -EFAULT;
+    }
+
+    if (fib_sr_check_lifetime(fib_sr, &now) == -ENOENT) {
+        mutex_unlock(&mtx_access);
+        return -ENOENT;
+    }
+
+    fib_sr_entry_t *elt;
+    LL_FOREACH(fib_sr->sr_path, elt) {
+        size_t addr_size_match = addr_size << 3;
+        if (universal_address_compare(elt->address, addr, &addr_size_match) == 0) {
+
+            /* temporary workaround to calm compiler */
+            (void)sr_path_entry;
+
+            *sr_path_entry = elt;
+            mutex_unlock(&mtx_access);
+            return 0;
+        }
+    }
+
+    mutex_unlock(&mtx_access);
+    return -EHOSTUNREACH;
+}
+
+int fib_sr_entry_append(fib_sr_t *fib_sr, uint8_t *addr, size_t addr_size)
+{
+    timex_t now;
+    vtimer_now(&now);
+
+    mutex_lock(&mtx_access);
+    if (fib_sr == NULL || addr == NULL) {
+        mutex_unlock(&mtx_access);
+        return -EFAULT;
+    }
+
+    if (fib_sr_check_lifetime(fib_sr, &now) == -ENOENT) {
+        mutex_unlock(&mtx_access);
+        return -ENOENT;
+    }
+
+    fib_sr_entry_t *elt;
+    LL_FOREACH(fib_sr->sr_path, elt) {
+        size_t addr_size_match = addr_size << 3;
+        if (universal_address_compare(elt->address, addr, &addr_size_match) == 0) {
+
+            mutex_unlock(&mtx_access);
+            return -EINVAL;
+        }
+    }
+
+    fib_sr_entry_t *new_entry[1];
+    int ret = fib_sr_new_entry(addr, addr_size, &new_entry[0]);
+
+    if (ret == 0) {
+        fib_sr_entry_t *tmp = fib_sr->sr_dest;
+        if (tmp != NULL) {
+            /* we append the new entry behind the former destination */
+            tmp->next = new_entry[0];
+        }
+        else {
+            /* this is also our first entry */
+            fib_sr->sr_path = new_entry[0];
+        }
+        fib_sr->sr_dest = new_entry[0];
+    }
+
+    mutex_unlock(&mtx_access);
+    return ret;
+}
+
+int fib_sr_entry_add(fib_sr_t *fib_sr, fib_sr_entry_t *sr_path_entry,
+                     uint8_t *addr, size_t addr_size, bool keep_remaining_route)
+{
+    timex_t now;
+    vtimer_now(&now);
+
+    mutex_lock(&mtx_access);
+    if (fib_sr == NULL || sr_path_entry == NULL || addr == NULL) {
+        mutex_unlock(&mtx_access);
+        return -EFAULT;
+    }
+
+    if (fib_sr_check_lifetime(fib_sr, &now) == -ENOENT) {
+        mutex_unlock(&mtx_access);
+        return -ENOENT;
+    }
+
+    bool found = false;
+    fib_sr_entry_t *elt;
+    LL_FOREACH(fib_sr->sr_path, elt) {
+        size_t addr_size_match = addr_size << 3;
+        if (universal_address_compare(elt->address, addr, &addr_size_match) == 0) {
+
+            mutex_unlock(&mtx_access);
+            return -EINVAL;
+        }
+        if (sr_path_entry == elt) {
+            found = true;
+        }
+    }
+
+    int ret = -ENOENT;
+    if (found) {
+        fib_sr_entry_t *new_entry[1];
```

Comment:
here again, you could drop the array declaration. Am I unaware of some fancy trick, or is it just a matter of taste?